packages/dds/map/src/mapKernel.ts

```diff
@@ -353,6 +370,38 @@ export class MapKernel {
 		this.submitMapClearMessage(op, copy);
 	}
 
+	/**
+	 * Get the attribution about the target key
+	 * @param key - Key to track
+	 * @returns The Attribution Key if the attribution tracking is enabled and key exists, otherwise undefined
+	 * @alpha
+	 */
+	public getAttribution(key: string): AttributionKey | undefined {
+		return this.attribution?.get(key);
+	}
+
+	/**
+	 * Get the whole attribution table
+	 * @returns The attribution table if the attribution tracking is enabled, otherwise undefined
+	 * @alpha
+	 */
+
+	public getAllAttribution(): Map<string, AttributionKey> | undefined {
+		return this.attribution;
+	}
+
+	private setAttribution(message: ISequencedDocumentMessage, key: string): void {
+		if (this.options?.attribution?.track ?? false) {
+			this.attribution?.set(key, { type: "op", seq: message.sequenceNumber });
```

Comment:
you should add an e2e test for how this behaves in simple attached and detached cases.

Reply:
@clarenceli-msft were you going to do this in a follow-up? (I'm going over my prior comments now before looking at the PR holistically)

Reply:
yes its covered in 5a36a84e02fda0998408017f12c251f2c238c20d